src/useObserver.ts

```diff
@@ -25,25 +25,17 @@ export function useObserver<T>(
     const forceUpdate = wantedForceUpdateHook()
 
     const reaction = useRef<Reaction | null>(null)
-    const committed = useRef(false)
-
     if (!reaction.current) {
-        // First render for this component. Not yet committed.
         reaction.current = new Reaction(`observer(${baseComponentName})`, () => {
-            // Observable has changed. Only force an update if we've definitely
-            // been committed.
-            if (committed.current) {
-                forceUpdate()
-            }
+            forceUpdate()
         })
     }
 
     useDebugValue(reaction, printDebugValue)
 
-    useEffect(() => {
```

Comment:
Changing this `useEffect` hook to `useLayoutEffect` fixes the bug for me.

Reply:
We certainly don't want to `useLayoutEffect`, that's bad for karma and stuff :) Besides, you said this PR alone fixed it, so I am not sure why would modify it even further. Please do provide an example, this is really hard to understand what kind of issue it actually is.

Reply:
ha, yeah, makes sense. Out of curiosity, what’s the reason that `committed.current` is set _asynchronously_ in the `useEffect` hook rather than _synchronously_ in the body of the `useObserver` hook?

Reply:
Moving `committed.current = true;` out of the `useEffect` hook also fixes the bug I was encountering.

Reply:
Don't worry about this. It was really just an early experiment. See #121 for a proper solution and some details.

Reply:
rad, thanks 🎉